src/main/java/ch/njol/skript/lang/SyntaxElementInfo.java

```diff
@@ -34,6 +35,9 @@ public SyntaxElementInfo(final String[] patterns, final Class<E> c, final String
 		this.patterns = patterns;
 		this.c = c;
 		this.originClassPath = originClassPath;
+		if (Modifier.isAbstract(c.getModifiers())) {
```

Comment:
Actually I'd say it's better to validate the parameters at the beginning of the constructor before assigning any fields